pytorch_lightning/accelerators/ddp_slurm_backend.py

```diff
@@ -0,0 +1,179 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+import os
+import torch
+import torch.distributed as torch_distrib
+import torch.distributed as dist
+
+from pytorch_lightning.accelerators.base_backend import Accelerator
+from pytorch_lightning import _logger as log
+from pytorch_lightning.utilities import AMPType
+from pytorch_lightning.utilities.distributed import rank_zero_only
+from pytorch_lightning.utilities.seed import seed_everything
+from pytorch_lightning.distributed.dist import LightningDistributed
+
+
+try:
+    from hydra.utils import to_absolute_path, get_original_cwd
+    from hydra.core.hydra_config import HydraConfig
+except ImportError:
+    HYDRA_AVAILABLE = False
+else:
+    HYDRA_AVAILABLE = True
+
+
+# -------------------------------------------
+# !!!!!!!!!!!!!! NOTE !!!!!!!!!!!!!!!!!!!!!!
+# TEMP CLASS WHILE WE DECOUPLE SLURM FROM DDP
+# !!!!!!!!!!!!!! NOTE !!!!!!!!!!!!!!!!!!!!!!
+# -------------------------------------------
+class DDPSLURMBackend(Accelerator):
```

Comment:
what's the final envisioned setup for slurm + DDP?

Reply:
te is already split out :)

I think right now it's to finish fixing DDP and having 3 types of ddp tied to the same file is making debugging impossible.

Once those tests pass, i'll fix the TE cpu ddp.

Then we'll see where we are with time, but i'd like to make something like a cluster_manager or something that is passed into the trainer as well and then linked to a backend.